common/src/str.rs

```diff
@@ -345,10 +351,7 @@ macro_rules! ascii {
 /// Get a Display-able type that formats to the python `repr()` of the string value
 #[inline]
 pub fn repr(s: &str) -> Repr<'_> {
```

Comment:
Should there be a `repr_with_quote` here to mirror the bytes API? Or is the intent that we construct a `Repr` directly?

Reply:
The latter, but the current state seems ugly. Is it better to make advance now?

The function `repr` seems not doing much thing. Probably entirely removing it will be better. I now think doing this right now will be better.

In future, After renaming `Repr` to `Escape`, `Escape::new(quote, prefix, suffix)`, `Escape::with_single()`, `Escape::with_quote()` etc seems better way than having many functions. Could you share your expectation?

Reply:
Yeah I think something like what you're describing sounds reasonable. But I'm fine with anything in the interim, as long as the functionality is exposed in some sort of public interface.